### PR TITLE
chore: added upgrade handler for new token emission

### PR DESCRIPTION
**Pull request by gsk967**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/upgrades.go [hunk 1]
```diff
@@ -55,6 +56,24 @@ func (app UmeeApp) RegisterUpgradeHandlers(bool) {
...
+	// TODO:finalize the name
+	planName := "token_emission"
+	app.UpgradeKeeper.SetUpgradeHandler(planName,
+		func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		},
+	)
+
+	app.storeUpgrade(planName, upgradeInfo, storetypes.StoreUpgrades{
+		Added:   []string{metoken.ModuleName},
```
Comment (toteki) [inline app/upgrades.go:74]:
```suggestion
		Added:   []string{/*metoken.ModuleName*/},
```
I think metoken module is still experimental flag, so if we merge this to main now, the added module can be commented out

**Review by toteki — APPROVED** [review]
Added approval, but let me know  on slack what our plans are for the testnet are

File: x/ugov/migrations/v2/store.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// MigrateStore performs in-place store migrations from 1 to 2
+func MigrateStore(ctx sdk.Context, key storetypes.StoreKey) error {
+	kvStore := ctx.KVStore(key)
+
+	ip := ugov.DefaultInflationParams()
+	if err := store.SetValue(kvStore, ugov.KeyInflationParams, &ip, "inflation_params"); err != nil {
+		return err
+	}
+
+	cycleEnd := ctx.BlockTime().Add(ip.InflationCycle)
```
[the remark this answers is not in the document]
Comment (gsk967) [reply x/ugov/migrations/v2/store.go:19]: Yeah , I already checked this everything working fine 
`ctx.BlockTIme()` will return current block time only

File: app/upgrades.go (hunk 1 above)
Comment (gsk967) [reply app/upgrades.go:74]: I will comment the this upgrade handler

**Pull request merged**